src/payment/payment-methods/stripe/service.js

```diff
@@ -0,0 +1,109 @@
+import formurlencoded from 'form-urlencoded';
+
+import { ensureConfig, getConfig } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { logError } from '@edx/frontend-platform/logging';
+
+import { handleApiError } from '../../data/handleRequestError';
+
+ensureConfig(['ECOMMERCE_BASE_URL', 'STRIPE_RESPONSE_URL'], 'Stripe API service');
+
+/**
+ * Checkout with Stripe
+ *
+ * 1. Update Payment Intent with billing form data on submit
+ * 2. POST request to ecommerce Stripe API
+ * 3. Redirect to receipt page
+ */
+export default async function checkout(
+  basket,
+  {
+    cardHolderInfo, stripe, elements, context, products,
+  },
+  setLocation = href => { global.location.href = href; }, // HACK: allow tests to mock setting location
+) {
+  async function stripePaymentMethodHandler(result) {
+    if (result.error) {
+      // Show error in payment form
+      if (result.error.type === 'card_error' || result.error.type === 'validation_error') {
+        // setMessage(result.error.message);
```

Comment:
This is local state in the StripePaymentForm component which I don't have access here, console logging for now

Reply:
This will mean some client-stripe Stripe validation messaging won't show to the user, correct? If we're not immediately planning to address, could we ticket remaining work here?